docs/pages/guides/server-components.mdx

```diff
@@ -207,6 +207,98 @@ export default function Button({ title }) {
 - Server actions can only return serializable data.
 - Server Actions run on the server and are a good place to put logic that should not be exposed to the client.
 
+### Rendering in Server Actions
+
+Server Actions in Expo Router can render React components on the server and stream back an **RSC payload** (a custom JSON-like format that's maintained by the React team) for rendering on the client. This is similar to server-side rendering (SSR) on the web,
+
+For example, the following server action will render some text:
+
+```tsx components/server-actions.tsx
+'use server';
+
+// Optional: Import "server-only" for sanity.
+import 'server-only';
+
+import { View, Image, Text } from 'react-native';
+
+export async function renderProfile({
+  username,
+  accessToken,
+}: {
+  username: string;
+  accessToken: string;
+}) {
+  // NOTE: Rate limits, GDPR, and other server-side operations can be done here.
+
+  // Fetch some data securely from an API.
+  const { name, image } = await fetch(`https://api.example.com/profile/${username}`, {
+    headers: {
+      Authorization: `Bearer ${accessToken}`,
+      // Use secret environment variables securely as this code will live on the server.
+      // The EXPO_PUBLIC_ prefix is not required here.
+      'X-Secret': process.env.SECRET,
+    },
+  }).then(res => res.json());
+
+  // Render
+  return (
+    <View>
+      <Image source={{ uri: image }} />
+      <Text>{name}</Text>
+    </View>
+  );
+}
+```
+
+This server action can be invoked from a client component and the contents will be streamed back to the client:
+
+```tsx components/profile.tsx
+'use client';
+
+import { useLocalSearchParams } from 'expo-router';
+import * as React from 'react';
+import { Text } from 'react-native';
+
+import { renderProfile } from '@/components/server-actions';
+
+// Loading state that renders while data is being fetched.
+function Fallback() {
+  return <Text>Loading...</Text>;
+}
+
+export default function Profile() {
+  const { username } = useLocalSearchParams();
+  const { accessToken } = useCustomAuthProvider();
+
+  // Call the server action with the username and access token.
+  const profile = React.useMemo(
+    () => renderProfile({ username, accessToken }),
+    [username, accessToken]
+  );
+
+  // Render the profile asynchronously with React suspense and a custom loading state.
+  return <React.Suspense fallback={<Fallback />}>{profile}</React.Suspense>;
+}
+```
+
+## Library compatibility
+
+Not all libraries are optimized for React Server Components yet. You can use the `"use client"` directive to mark a file as a client component and use it in a server component, this can be used to temporarily workaround compatibility issues.
```

Comment:
```suggestion
Not all libraries are optimized for React Server Components yet. You can use the `"use client" directive to mark a file as a Client Component and use it in a Server Component. This can be used to temporarily workaround compatibility issues.
```